cmd/apiserver/app/server/util.go

```diff
@@ -17,48 +17,154 @@ limitations under the License.
+		sharedInformers: sharedInformers,
 	}
+	if inCluster {
+		inClusterConfig, err := restclient.InClusterConfig()
+		if err != nil {
+			glog.Errorf("Failed to get kube client config: %v", err)
+			return nil, nil, err
+		}
+		inClusterConfig.GroupVersion = &schema.GroupVersion{}
+
+		kubeClient, err := kubeclientset.NewForConfig(inClusterConfig)
+		if err != nil {
+			glog.Errorf("Failed to create clientset interface: %v", err)
+			return nil, nil, err
+		}
 
-	// glog.V(4).Info("Setting up authn (disabled)")
-	// need to figure out what's throwing the `missing clientCA file` err
-	/*
-		if _, err := genericConfig.ApplyDelegatingAuthenticationOptions(serverOptions.AuthenticationOptions); err != nil {
-			glog.Infoln(err)
-			return err
+		kubeSharedInformers := kubeinformers.NewSharedInformerFactory(kubeClient, 10*time.Minute)
+
+		// TODO: we need upstream to package AlwaysAdmit, or stop defaulting to it!
+		// NOTE: right now, we only run admission controllers when on kube cluster.
+		genericConfig.AdmissionControl, err = buildAdmission(s, client, sharedInformers, kubeClient, kubeSharedInformers)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to initialize admission: %v", err)
 		}
-	*/
-
-	// glog.V(4).Infoln("Setting up authz (disabled)")
-	// having this enabled causes the server to crash for any call
-	/*
-		if _, err := genericConfig.ApplyDelegatingAuthorizationOptions(serverOptions.AuthorizationOptions); err != nil {
-			glog.Infoln(err)
-			return err
+
+		scConfig.kubeClient = kubeClient
+		scConfig.kubeSharedInformers = kubeSharedInformers
+	}
+
+	return genericConfig, scConfig, nil
+}
+
+// buildAdmission constructs the admission chain
+func buildAdmission(s *ServiceCatalogServerOptions,
+	client internalclientset.Interface, sharedInformers informers.SharedInformerFactory,
+	kubeClient kubeclientset.Interface, kubeSharedInformers kubeinformers.SharedInformerFactory) (admission.Interface, error) {
+
+	admissionControlPluginNames := strings.Split(s.GenericServerRunOptions.AdmissionControl, ",")
+	glog.Infof("Admission control plugin names: %v", admissionControlPluginNames)
+	var err error
+
+	pluginInitializer := scadmission.NewPluginInitializer(client, sharedInformers, kubeClient, kubeSharedInformers)
+	admissionConfigProvider, err := admission.ReadAdmissionConfiguration(admissionControlPluginNames, s.GenericServerRunOptions.AdmissionControlConfigFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read plugin config: %v", err)
+	}
+	return admission.NewFromPlugins(admissionControlPluginNames, admissionConfigProvider, pluginInitializer)
+}
+
+// addPostStartHooks adds the common post start hooks we invoke when using either server storage option.
+func addPostStartHooks(server *genericapiserver.GenericAPIServer, scConfig *serviceCatalogConfig, stopCh <-chan struct{}) {
+	server.AddPostStartHook("start-service-catalog-apiserver-informers", func(context genericapiserver.PostStartHookContext) error {
```

Comment:
@MHBauer Unfortunately, I think your https://github.com/kubernetes/community/pull/309 PR is the document :(